### DISCUSSION TITLE
can pyright display the "full module paths" in error messages?

### DISCUSSION THREAD

**Discussion by Sorontik** [microsoft/pyright#5406] [Q&A] [answered]
Hi,

I was wondering if one can tell pyright to display the "full name" of types in error messages, like `datetime.timezone` instead of just `timezone`.

Motivation:
I use a library that basically provides a container-class and a default element-class.
When creating a new container, a custom subclass of that elementclass can be passed on to extend functionality.

The default elementclass uses the obvious name "Elementclass" and when i created my custom class, i didn't know about the name of the default class and chose the intuitive "Elementclass" as well.

This left me with two classes with identical names in different modules (one of which is not even in my current project) and eventually (inevitably) i got them mixed up.

The main Problem was that pyrights error message told me that
`Type "Elementclass" cannot be assigned to type "Elementclass"`
and i took forever (longer than it should have) for me to understand that these refer to two different types with the same name.

This would have been obvious, if the error message had been
`Type "library.Elementclass" cannot be assigned to type "mymodule.Elementclass"`

**Comment by erictraut** [comment, marked as answer]
Pyright doesn't normally report the full names of types. In the vast majority of cases, this is unnecessary and would make error messages more verbose and difficult to read. In cases where two type names overlap, pyright does use the full name. In response to your question, I reviewed the code and found a few additional places where full names could be used conditionally, so I've updated those cases. This will be included in the next release.
